server/test/connected_objects_test.go

```diff
@@ -0,0 +1,671 @@
+				User:       &openfgapb.ObjectRelation{Object: "user:jon"},
+			},
+			resolveNodeLimit: 2,
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "folder",
+						Relations: map[string]*openfgapb.Userset{
+							"parent": {Userset: &openfgapb.Userset_This{}},
+							"viewer": {
+								Userset: &openfgapb.Userset_Union{
+									Union: &openfgapb.Usersets{
+										Child: []*openfgapb.Userset{
+											{Userset: &openfgapb.Userset_This{}},
+											{Userset: &openfgapb.Userset_TupleToUserset{
+												TupleToUserset: &openfgapb.TupleToUserset{
+													Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+													ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+												},
+											}},
+										},
+									},
+								},
+							},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "folder"},
+									},
+								},
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:folder2", "parent", "folder:folder1"),
+				tuple.NewTupleKey("folder:folder3", "parent", "folder:folder2"),
+			},
+			expectedError: serverErrors.AuthorizationModelResolutionTooComplex,
+		},
+		{
+			name: "Test-10",
+			request: &commands.ConnectedObjectsRequest{
+				StoreID:    store,
+				ObjectType: "group",
+				Relation:   "member",
+				User:       &openfgapb.ObjectRelation{Object: "group:iam", Relation: "member"},
+			},
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "group",
+						Relations: map[string]*openfgapb.Userset{
+							"member": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"member": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "group", Relation: "member"},
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("group:auth0", "member", "group:eng#member"),
+				tuple.NewTupleKey("group:eng", "member", "group:iam#member"),
+				tuple.NewTupleKey("group:iam", "member", "user:jon"),
+			},
+			expectedObjects: []string{"group:auth0", "group:eng"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
```

Comment:
I'm guessing this is WIP?

Reply:
Oh, good find. That's just residual. It should be wired up to a datastore input at the top-level of the function. Will do what we've done elsewhere here with

```
func ConnectedObjectsTest(t *testing.T, ds storage.OpenFGADatastore) {
    ...
    _ = ds // use ds
}
```

Reply:
Resolved.